pkgs/development/libraries/symengine/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv
+, fetchFromGitHub
+, cmake
+, gmp
+, flint
+}:
+
+stdenv.mkDerivation rec {
+  name = "symengine-${version}";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "symengine";
+    repo = "symengine";
+    rev = "v${version}";
+    sha256 = "1p7hk163rgn1zzvjlq4vskblry3s2rg5bc7xlr08wfqckfr47bqc";
+  };
+
+  buildInputs = [ cmake gmp flint ];
+
+  cmakeFlags = [
+    "-DWITH_FLINT=ON"
```

Comment:
In addition to this, use
```
    -DINTEGER_CLASS=flint
    -DWITH_SYMENGINE_THREAD_SAFE=yes
    -DWITH_MPC=yes
    -DWITH_LLVM=yes
    -DBUILD_FOR_DISTRIBUTION=yes
    -DBUILD_SHARED_LIBS=yes
```

and add dependencies on `mpfr, mpc, llvm`. (If you package symengine 0.4, you can use LLVM 3.8-8.0, if it's 0.3, LLVM 3.8-4.0. If LLVM is not in nix, you can skip it.) MPFR and MPC are kind of important dependencies (even though they are optional)

Reply:
Thanks for the advice! I will definitely be packing symengine this way. Yes we do have llvm as well. Currently I use it a lot with `numba` in nix.

Reply:
You probably need `-DBUILD_FOR_DISTRIBUTION=yes` to avoid adding `-march=native` by default. (This flag will probably be removed from default flags in the future.) You also need `-DBUILD_SHARED_LIBS=yes` for shared libraries.